src/app/components/searchbar/searchbar.component.ts

```diff
@@ -1,8 +1,26 @@
 import { Component } from '@angular/core';
+import { TreeModel, TreeNode } from '@circlon/angular-tree-component';
+import { SharedTreeDataService } from 'src/app/services/shared-tree-data.service';
+import { fuzzySearch } from 'src/app/utils';
+
+interface SelectedValues {
+  selectedValues: string[];
+}
 
 @Component({
   selector: 'app-searchbar',
   templateUrl: './searchbar.component.html',
   styleUrls: ['./searchbar.component.scss'],
 })
-export class SearchbarComponent {}
+export class SearchbarComponent {
+  public tree: { treeModel: TreeModel };
+
+  constructor(private sharedDataSvc: SharedTreeDataService) {}
+
+  /* will filter the tree and only display nodes that match selected values */
+  filterTree = ({selectedValues}: SelectedValues) => () => {
+    this.sharedDataSvc.tree.treeModel.filterNodes((node: TreeNode) =>
+      selectedValues.length === 0 ? true : fuzzySearch(selectedValues, node.data.name)
+    );
```

Comment:
Maybe this can be addressed later, but I think this shouldn't filter out the subtree of a node that is matched. So, if you search for Fruit then the tree should contain the Fruit tag and the whole list of fruits (Apple, Strawberry, etc.) under it.

Reply:
Yeah, I agree. I didn't think about how we would want to see the subtree. Initially, I'm not sure how to do that but I'm sure there is a way if we change the logic here or in the fuzzySearch().